libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -693,21 +710,92 @@ def bind_tools(
         """
 
         formatted_tools = [convert_to_openai_tool(tool) for tool in tools]
-        if tool_choice is not None:
-            if isinstance(tool_choice, str) and (tool_choice not in ("auto", "none")):
-                tool_choice = {"type": "function", "function": {"name": tool_choice}}
-            if isinstance(tool_choice, dict) and (len(formatted_tools) != 1):
+        if tool_choice is not None and tool_choice:
+            if len(formatted_tools) != 1:
                 raise ValueError(
                     "When specifying `tool_choice`, you must provide exactly one "
                     f"tool. Received {len(formatted_tools)} tools."
                 )
-            if isinstance(tool_choice, dict) and (
-                formatted_tools[0]["function"]["name"]
-                != tool_choice["function"]["name"]
-            ):
+            if isinstance(tool_choice, str):
+                if tool_choice not in ("auto", "none"):
+                    tool_choice = {
+                        "type": "function",
+                        "function": {"name": tool_choice},
+                    }
+            elif isinstance(tool_choice, bool):
+                tool_choice = formatted_tools[0]
+            elif isinstance(tool_choice, dict):
+                if (
+                    formatted_tools[0]["function"]["name"]
+                    != tool_choice["function"]["name"]
+                ):
+                    raise ValueError(
+                        f"Tool choice {tool_choice} was specified, but the only "
+                        f"provided tool was {formatted_tools[0]['function']['name']}."
+                    )
+            else:
                 raise ValueError(
-                    f"Tool choice {tool_choice} was specified, but the only "
-                    f"provided tool was {formatted_tools[0]['function']['name']}."
+                    f"Unrecognized tool_choice type. Expected str, bool or dict. "
+                    f"Received: {tool_choice}"
                 )
             kwargs["tool_choice"] = tool_choice
         return super().bind(tools=formatted_tools, **kwargs)
+
+    def with_output_format(
+        self,
+        output_schema: _OutputSchema,
+        *,
+        method: Literal["function_calling", "json_format"] = "function_calling",
+        return_type: Literal["raw", "parsed", "all"] = "parsed",
+        **kwargs: Any,
+    ) -> Runnable[LanguageModelInput, _OutputFormat]:
+        """"""
+        if kwargs:
+            raise ValueError(f"Received unsupported arguments {kwargs}")
+        is_pydantic_schema = _is_pydantic_class(output_schema)
```

Comment:
I am still in favor of making a conceptual distinction between schema declaration vs. the parsing.

For example, I will usually want to declare my schema with pydantic, but I don't necessarily want the output to be in pydantic (e.g., i want to stream the structured json out)

Reply:
you can always `dict_schema = convert_to_openai_tool(pydantic_schema)` before passing in schema? i just don't like adding a parameter until we know it's really needed. we can always add it later

Reply:
OK that will work. We can add a `validator='auto'" parameter later on, allowing one to set `validator=None` or `validator=Type[BaseModel]` or something along those lines?